scripts/configurator/parser/cmake_var.py

```diff
@@ -1,9 +1,108 @@
-from typing import Dict, Any, Type, Mapping
+from typing import Dict, Any, Optional, Type, Mapping, List
+import re
+from sys import stderr
+
+
+class metadata:
+   keyword_expr = re.compile(r"([A-Z_]+)")
+
+   def __init__(self, comment: List[str]) -> None:
+      self.depends_on: Optional[List[str]] = None
+      self.group: Optional[str] = None
+      self.realtype: Optional[str] = None
+
+      self.kw_to_parse: Mapping[str, function] = {
+         "REALTYPE" : self.parse_realtype,
+         "DEPENDS_ON" : self.parse_depends_on,
+         "GROUP" : self.parse_group
+      }
+
+      self.kw_to_setter = {
+         "REALTYPE": self.get_setter_realtype(),
```

Comment:
with the generic setter, we'll have to pass here the name of the variable, but that's a good enough trade-off to avoid the boilerplate setter for each metadata field.

Reply:
changed